transcom/mymove
# B-20647-INT add sac 80 character limit

## [B-20647](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A982736)

## Summary

Set the SAC field character limit to 80 characters for HHG and NTS shipments.

Is there anything you would like reviewers to give additional scrutiny?
Also set backend validation so the SAC cannot be more than 80 characters. Please test using swagger by trying to create a a new Order. See instructions in test section.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access a shipment with HHG and/or NTS as Service Counselor that hasn't been approved yet
2. View the Orders for the HHG and/or NTS shipment
3. Try to enter more than 80 characters into the SAC field. Below is a sample string that is 90 characters, if you paste this into the SAC field then you should not see any 9's in the text field.

111111111122222222223333333333444444444455555555556666666666777777777788888888889999999999

4. Use swagger to try and create an Order with a SAC greater than 80 characters. This should fail with an unprocessable entity error. 
5. Adjust the SAC to be less than 80 characters and verify you get no errors.

Sample for the body for POST create Order is below just update the SAC as needed.

{
  "serviceMemberId": "eec4e65e-b328-4133-b822-c7eaba860ec9",
  "issueDate": "2024-07-22",
  "reportByDate": "2024-07-22",
  "ordersType": "PERMANENT_CHANGE_OF_STATION",
  "ordersTypeDetail": "HHG_PERMITTED",
  "hasDependents": true,
  "spouseHasProGear": true,
  "newDutyLocationId": "62e795f4-61a8-4cca-b59d-028345c07b89",
  "ordersNumber": "030-00362",
  "tac": "F8J1",
  "sac": "111111111122222222223333333333444444444455555555556666666666777777777788888888889999999999",
  "departmentIndicator": "NAVY_AND_MARINES",
  "grade": "E_1",
  "originDutyLocationId": "929ba47a-2f7d-430e-9655-157a4c80303d"
}

Sample body for the PATCH update of the order (use the eTag and id from the successful result above.

{
  "issueDate": "2024-07-22",
  "reportByDate": "2024-07-22",
  "ordersType": "PERMANENT_CHANGE_OF_STATION",
  "ordersTypeDetail": "HHG_PERMITTED",
  "newDutyLocationId": "62e795f4-61a8-4cca-b59d-028345c07b89",
  "ordersNumber": "030-00362",
  "tac": "F8J1",
  "sac": "11111111112222222222333333333344444444445555555555666666666",
  "departmentIndicator": "NAVY_AND_MARINES",
  "grade": "E_1",
  "originDutyLocationId": "929ba47a-2f7d-430e-9655-157a4c80303d"
}

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
